Vostok.Applications.AspNetCore/IApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Options;
+using Vostok.Applications.AspNetCore.Configuration;
+using Vostok.Applications.AspNetCore.Middlewares;
+using Vostok.Throttling;
+
+namespace Vostok.Applications.AspNetCore
+{
+    [PublicAPI]
+    public static class IApplicationBuilderExtensions
+    {
+        #region Distributed context
+
+        /// <summary>
+        /// Adds the <see cref="DistributedContextMiddleware"/> to the specified application builder.
+        /// </summary>
+        public static IApplicationBuilder UseVostokDistributedContext(this IApplicationBuilder builder)
+            => builder.UseMiddleware<DistributedContextMiddleware>();
+
+        /// <summary>
+        /// Adds the <see cref="DistributedContextMiddleware"/> to the specified application builder.
+        /// </summary>
+        public static IApplicationBuilder UseVostokDistributedContext(this IApplicationBuilder builder, Action<DistributedContextSettings> setup)
+            => builder.UseMiddleware<DistributedContextMiddleware>(Configure(setup));
```

Comment:
Most common way of doing this is to create extension for `IServiceCollection` like `AddVostokDistributedContext<T>(Action<T>)` which will internally use `IServiceCollection.Configure<T>(Action<T>)`. 
Such approach allows to utilize `IPostConfigureOptions` capabilities, overriding configuration later in `ConfigureServices` section of Startup & saves options in DI. 

This overload mostly used in cases when ctor parameter has common time (e.g. UseCors(policyName))